state-chain/pallets/cf-elections/src/electoral_systems/solana_swap_accounts_tracking.rs

```diff
@@ -0,0 +1,214 @@
+		Account,
+		SwapDetails,
+		BlockNumber,
+		Settings,
+		Hook,
+		ValidatorId,
+		E,
+	)>,
+}
+impl<
+		E: sp_std::fmt::Debug + 'static,
+		Account: MaybeSerializeDeserialize + Member + Parameter + Ord,
+		SwapDetails: MaybeSerializeDeserialize + Member + Parameter + Ord,
+		BlockNumber: MaybeSerializeDeserialize + Member + Parameter + Ord + CheckedSub + Into<u32> + Copy,
+		Settings: Member + Parameter + MaybeSerializeDeserialize + Eq,
+		Hook: SolanaVaultSwapAccountsHook<Account, SwapDetails, E> + 'static,
+		ValidatorId: Member + Parameter + Ord + MaybeSerializeDeserialize,
+	> ElectoralSystem
+	for SolanaVaultSwapAccounts<Account, SwapDetails, BlockNumber, Settings, Hook, ValidatorId, E>
+{
+	type ValidatorId = ValidatorId;
+	type ElectoralUnsynchronisedState = SolanaVaultSwapsElectoralState<Account, BlockNumber>;
+	type ElectoralUnsynchronisedStateMapKey = ();
+	type ElectoralUnsynchronisedStateMapValue = ();
+
+	type ElectoralUnsynchronisedSettings = ();
+	type ElectoralSettings = Settings;
+	type ElectionIdentifierExtra = ();
+	type ElectionProperties = ();
+	type ElectionState = ();
+	type Vote = vote_storage::bitmap::Bitmap<SolanaVaultSwapsVote<Account, SwapDetails>>;
+	type Consensus = SolanaVaultSwapsVote<Account, SwapDetails>;
+	type OnFinalizeContext = BlockNumber;
+	type OnFinalizeReturn = ();
+
+	fn generate_vote_properties(
+		_election_identifier: ElectionIdentifier<Self::ElectionIdentifierExtra>,
+		_previous_vote: Option<(VotePropertiesOf<Self>, AuthorityVoteOf<Self>)>,
+		_vote: &<Self::Vote as VoteStorage>::PartialVote,
+	) -> Result<VotePropertiesOf<Self>, CorruptStorageError> {
+		Ok(())
+	}
+
+	fn on_finalize<ElectoralAccess: ElectoralWriteAccess<ElectoralSystem = Self>>(
+		electoral_access: &mut ElectoralAccess,
+		election_identifiers: Vec<ElectionIdentifier<Self::ElectionIdentifierExtra>>,
+		current_block_number: &Self::OnFinalizeContext,
+	) -> Result<Self::OnFinalizeReturn, CorruptStorageError> {
+		if let Some(election_identifier) = election_identifiers
+			.into_iter()
+			.at_most_one()
+			.map_err(|_| CorruptStorageError::new())?
+		{
+			let mut election_access = electoral_access.election_mut(election_identifier)?;
+			if let Some(consensus) = election_access.check_consensus()?.has_consensus() {
+				election_access.delete();
+				electoral_access.new_election((), (), ())?;
+				electoral_access.mutate_unsynchronised_state(|_, unsynchronised_state| {
+					unsynchronised_state.witnessed_open_accounts.extend(
+						consensus.new_accounts.iter().map(|(account, swap_details)| {
+							Hook::initiate_vault_swap((*swap_details).clone());
+							(*account).clone()
+						}),
+					);
+
+					consensus.confirm_closed_accounts.into_iter().for_each(|acc| {
+						unsynchronised_state.closure_initiated_accounts.remove(&acc);
+					});
+
+					Ok(())
+				})?;
+			}
+		} else {
+			electoral_access.new_election((), (), ())?;
+		}
+
+		let mut unsynchronised_state = electoral_access.unsynchronised_state()?;
```

Comment:
When we're in specific implementations we can use more specific and descriptive variable names than "unsynchronised state" etc.